### PR TITLE
fix: task 1

### PR DESCRIPTION
**Pull request by tkachenya41**

### CODE REVIEW

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -69,10 +66,9 @@ function getStringFromTemplate(firstName, lastName) {
  *   'Hello, Chuck Norris!' => 'Chuck Norris'
  */
 function extractNameFromTemplate(value) {
-    throw new Error('Not implemented');
+  return value.match(/Hello, (.+?)!/)[1];
```
Comment (ak-krivonosov) [inline task/01-strings-tasks.js:69]: Why do you need a question mark?

File: task/01-strings-tasks.js [hunk 2]
```diff
@@ -84,7 +80,7 @@ function extractNameFromTemplate(value) {
  *   'cat'       => 'c'
  */
 function getFirstChar(value) {
-    throw new Error('Not implemented');
+  return value.charAt(0);
```
Comment (ak-krivonosov) [inline task/01-strings-tasks.js]: You can just use value[0], it’s shorter or if you need a char from the end use .at()

File: task/01-strings-tasks.js (hunk 1 above)
Comment (tkachenya41) [reply task/01-strings-tasks.js:69]: I read that this question mark choose the minimum of the words to satisfy pattern

File: task/01-strings-tasks.js (hunk 2 above)
Comment (tkachenya41) [reply task/01-strings-tasks.js]: Okay, thanks, I'll correct

File: task/01-strings-tasks.js [hunk 3]
```diff
@@ -201,9 +196,23 @@ function extractEmails(str) {
  *
  */
 function getRectangleString(width, height) {
-    throw new Error('Not implemented');
-}
+  let rectangle = "";
 
+  for (let i = 0; i < height; i++) {
+    if (i === 0) {
+      // Верхний ряд
```
Comment (ak-krivonosov) [inline task/01-strings-tasks.js]: Please always use English

File: task/01-strings-tasks.js [hunk 4]
```diff
@@ -201,9 +196,23 @@ function extractEmails(str) {
  *
  */
 function getRectangleString(width, height) {
-    throw new Error('Not implemented');
-}
+  let rectangle = "";
```
Comment (ak-krivonosov) [inline task/01-strings-tasks.js]: Please, try to solve it without using let

Comment (tkachenya41) [reply task/01-strings-tasks.js]: Sorry? Don't get it

Comment (ak-krivonosov) [reply task/01-strings-tasks.js]: You used here `let` for the rectangle variable, try to rewrite this function with `const`

**Review by ak-krivonosov — CHANGES_REQUESTED** [review]
Left some comments

File: task/01-strings-tasks.js [hunk 5]
```diff
@@ -200,10 +195,23 @@ function extractEmails(str) {
...
-    throw new Error('Not implemented');
-}
+const getRectangleString = (width, height) => {
+  const topRow = "┌" + "─".repeat(width - 2) + "┐\n";
+  const middleRow = "│" + " ".repeat(width - 2) + "│\n";
+  const bottomRow = "└" + "─".repeat(width - 2) + "┘\n";
+
+  let rectangle = topRow;
+
+  for (let i = 1; i < height - 1; i++) {
```
Comment (ak-krivonosov) [inline task/01-strings-tasks.js]: Try to replace this loop with `.repeat` method

File: task/01-strings-tasks.js [hunk 6]
```diff
@@ -200,10 +195,23 @@ function extractEmails(str) {
...
+    rectangle += middleRow;
+  }
 
+  if (height > 1) {
```
Comment (ak-krivonosov) [inline task/01-strings-tasks.js]: You don’t need this condition, because height is always bigger than 1

**Review by ak-krivonosov — CHANGES_REQUESTED** [review]
2 comments

**Review by ak-krivonosov — APPROVED** [review]

**Pull request merged**